doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,319 @@
+     - Add support to mirror both ingress/egress traffic on port/port-channel to SPAN/ERSPAN mirror session.
+
+2. Dynamic session management
+    - Allow multiple source to single destination.
+    - Each session supports mirroring from single port to single destination port.
+    - Session-id created in SAI per destination port will be used when the same destination port is configured in other session.
+      This effectively utilizes the hardware resource to be shared across multiple sessions.
+    - Mirror session on source portchannel will be active if at least one port is part of portchannel.
+    - Mirror session on source portchannel will become inactive when portchannel has no members.
+    - ERSPAN session will be active/inactive based on destination IP reachability.
+
+3. ACL rules can continue to use port/ERSPAN sessions as the action.
+
+4. Configuration CLI for mirror session
+    - CLI allows all flavors of mirror sessions.
+    - CLI validation for all mandatory parameters in ERSPAN configuration.
+    - CLI validation for all mandatory parameters in port/portchannel mirroring.
+    - CLI to allow mirror session configuration only with destination port.
+
+
+## 1.2 Configuration and Management Requirements
+- Existing CLI 'config mirror_session add/remove'to be extended to include source port/portchannel.
+- Existing CLI 'config mirror_session add/remove' to be extended to include destination port/portchannel.
+- Existing CLI 'show mirror session' is extended to support all flavors of mirror sessions.
+
+
+## 1.3 Scalability Requirements
+- Up to max ASIC capable mirror sessions to be supported.
+- Once max mirror sessions are created and user attempts to create new session, error will be logged in syslog.
+
+
+## 1.4 Warm Boot Requirements
+- Mirroring functionality should continue to work across warm reboot.
+
+To support planned system warm boot.
+To support SWSS docker warm boot.
+
+
+# 2 Functionality
+
+Refer section 1
+
+## 2.2 Functional Description
+Refer section 1.1
+
+## 2.3 Functional Description
+
+Mirroring to destination VLAN (RSPAN) is not supported in this release.
+
+# 3 Design
+## 3.1 Overview
+
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+
+Existing table PORT_MIRROR_TABLE is enhanced to accept new source and destination configuration options  in the configuration database. This table is filled by the management framework.
+
+#### CONFIG_PORT_MIRROR_TABLE
+
+    ;Configure SPAN/ERSPAN mirror session.
+    ;storm control type - broadcast / unknown-unicast / unknown-multicast
+    key       = PORT_MIRROR_TABLE:mirror_session_name ; mirror_session_name is
+                                                      ; unique session
+                                                      ; identifier
+    ;field  = value
+    destination_port = PORT_TABLE:ifname    ; ifname must be unique across PORT TABLE.
+    source_port = PORT_TABLE:ifname    ; ifname must be unique across PORT,INTF,LAG TABLES
```

Comment:
what do you mean by INTF? INTF is layer 3 concept.

Reply:
Doc is already updated. this is only port for destination port, port/LAG for src port.